onnx/defs/schema.h

```diff
@@ -1142,6 +1142,11 @@ class OpSchemaRegistry final : public ISchemaRegistry {
         bool fail_duplicate_schema = true) {
       ONNX_TRY {
         op_schema.Finalize();
+
+        // Acquires lock to thread-guard schema map access
+        auto* registry = OpSchemaRegistry::Instance();
```

Comment:
I think every time the map returned by OpSchemaRegistry::Instance() is accessed should be protected against concurrent accesses. Function `GetMapWithoutEnsuringRegistration` is called by another function not protected by the mutex. Maybe the datarace comes from it. Maybe it would be safer to create a specific class to hold the map storing the schemas and protect the accesses to this class instead of looking to every place this container is accessed and protect it.

Reply:
Where is the function called not protected by the mutex? I assume that this is a dynamic race detection in the given unit-test, where only registration/deregistration are called in parallel, both of which seem to invoke the function under a mutex?

Reply:
`GetMapWithoutEnsuringRegistration` is used in other places to access the registered schemas. If a schema is removed while another function is counting the number of schemas, it could lead to some unstable state. It is unlikely to happen but it could happen in a mutlithread scenarios. Every call to GetMapWithoutEnsuringRegistration should be protected with a lock.